### ISSUE TITLE
Git - display incoming/outgoing commits in the Source Control view container

### ISSUE THREAD

**Issue by lszomoru** [microsoft/vscode#192290]

**Comment by tusharsnx** [comment]
Thanks for adding this feature right inside Vscode 😄 I've been using Gitlens just to have this for a while now.

I think the view only shows changes/commits for the current branch. I'd appreciate if we could have a drop down menu for selecting the branch also, so it can show the changes from other branches as well, and not having to checkout branches for the changes to show. The selected branch could be remembered for the entire session, and resets after closing/reopening Vscode.

**Comment by tusharsnx** [comment]
As a contributor, many people work on a cloned repo, that has multiple tracking branches to a remote branch on that repo. But, in most cases we want to compare against the original repo's `main` (or any branch for that matter) where this change will be merged ultimately. Giving an optional "compare against" (along with 'Incoming Changes', 'Outgoing Changes') list item in the view would help in this case.

Current workaround is to use `git branch -u original/main local-branch` which then makes the 'Outgoing Changes' compare against `original/main`. However, this will make `git push/pull` more verbose as you'd have to use cloned remote name with the command every time.

**Comment by Wladefant** [comment]
I am a bit new to the party how to I enable this setting?

**Comment by gjsjohnmurray** [comment]
@Wladefant at the moment you will need to be using Insiders

https://code.visualstudio.com/insiders

**Comment by Wladefant** [comment]
Ok not that new. I have insiders installed, but where do I enable this setting?

**Comment by jcapcik-np** [comment]
Yes!  As a former Sourcetree user before switching to a Windows WSL env, this was a feature I desperately missed.  Thank you!

**Comment by aosar** [comment]
I am not a fan of this personally, so Im glad I found the setting to turn off. I was scared I did something wrong the first time I saw this.

Additionaly, I don't like the "Sync Changes" button so dangerously close to the commit menu, and in place of the commit button. There a lot of changes I like to keep local or that I want to redo, rebase, cherry-pick, etc before pushing out. I dont want to accidentally hit that button and cause chaos with the history rewriting, being forced to force push to fix it. Not to mention potential security concerns (such as committing a secret that's __only__ meant for your machine).
That could also result in collaborative workflow disruptions. If somebody pulls your accidental change that is not ready to use, then they have to reach out to you, and if you force push a fix then you have to work with them to get back in sync, and it's just a mess.

**Comment by lszomoru** [comment]
@aosar, thank you very much for the feedback. The "action button" in the Source Control view is meant to help with making the "next action" easily accessible. After one commits changes, the button switches to "Sync Changes" as that is the next natural action.

You can control the action button using the following settings:
* scm.showActionButton - completely disable the action button
* git.showActionButton - disable action button individually (commit, publish, sync)

**Comment by gjsjohnmurray** [comment]
> * I personally found this quite jarring to see this come up in the latest update, and had to Google how to hide it...

I recommend checking the release notes after an update. This section is one of the headline items for 1.85

https://code.visualstudio.com/updates/v1_85#_source-control

**Comment by lszomoru** [comment]
@benwinding, thank you very much for the feedback. Making toggling the visibility of the incoming/outgoing section has come up multiple times but I simply did not get to it. Having said that, I am planning on adding that in the upcoming release.

**Comment by cary-hu** [comment]
Can we mark the new feature disabled by default. I spend too much time to found how to hide this.

**Comment by eduard-sb** [comment]
This feature currently only shows **changes** against the upstream branch, however if the commits on the current branch have introduced new files they won't show up. It'd be cool if they did!

**Comment by lszomoru** [comment]
@eduard-sb, thanks for the feedback. Could you please file a separate issue so that I can track this. Providing a recording would also be really valuable for my to track down the root cause. Thanks!

**Comment by Daniel-Griffiths** [comment]
> ```json
>     "scm.showIncomingChanges" : false,
>     "scm.showOutgoingChanges" : false,
> ```

Not sure if something has changed since, but I had the use the string `"never"` instead of `false`

```json
  "scm.showIncomingChanges": "never",
  "scm.showOutgoingChanges": "never",
```
